src/lightning/fabric/plugins/environments/kubeflow.py

```diff
@@ -41,12 +44,8 @@ def main_port(self) -> int:
 
     @staticmethod
     def detect() -> bool:
-        """Returns ``True`` if the current process was launched using Kubeflow PyTorchJob."""
-        required_env_vars = {"KUBERNETES_PORT", "MASTER_ADDR", "MASTER_PORT", "WORLD_SIZE", "RANK"}
-        # torchelastic sets these. Make sure we're not in torchelastic
-        excluded_env_vars = {"GROUP_RANK", "LOCAL_RANK", "LOCAL_WORLD_SIZE"}
-        env_vars = os.environ.keys()
-        return required_env_vars.issubset(env_vars) and excluded_env_vars.isdisjoint(env_vars)
+        # The Kubeflow environment can't be detected automatically
+        return False
```

Comment:
We could also raise an exception here instead of returning False.

Reply:
Exception would be nicer...

Reply:
done :)